### PR TITLE
New settings for disable comments and social

### PR DESCRIPTION
**Pull request by FherStk**
Related with: https://github.com/DMOJ/online-judge/pull/2298

Added new settings to disable social and comments sections (false by default):
image

Blog post example with social and comments enabled (by default):
image

Blog post example with social and comments disabled:
image

### CODE REVIEW

File: dmoj/settings.py [hunk 1]
```diff
@@ -161,6 +161,9 @@
 FONTAWESOME_CSS = '//maxcdn.bootstrapcdn.com/font-awesome/4.3.0/css/font-awesome.min.css'
 DMOJ_CANONICAL = ''
 
+DMOJ_DISABLE_COMMENTS = False
```
Comment (Xyene) [inline dmoj/settings.py:164]: I think it makes the code a little clearer (and removes the double-negatives) if we had these be `DMOJ_ENABLE_{COMMENTS,SOCIAL} = True`.

File: templates/problem/editorial.html [hunk 2]
```diff
@@ -37,9 +37,12 @@
             {% endif %}
         {% endwith %}
         {{ solution.content|markdown('solution', MATH_ENGINE)|reference|str|safe }}
-    </div>
-    <hr>
-    {% include "comments/list.html" %}
+    </div> 
+       
+    {% if not disable_comments %}
```
Comment (Xyene) [inline templates/problem/editorial.html:42]: The comments used to be inside the `<div/>`, but are now outside. Was this change intentional?

File: dmoj/settings.py (hunk 1 above)
Comment (FherStk) [reply dmoj/settings.py]: Done :)

File: templates/problem/editorial.html (hunk 2 above)
Comment (FherStk) [reply templates/problem/editorial.html]: If I'm not wrong, is outside the "div" in the current main branch from where I branched:
https://github.com/DMOJ/online-judge/blob/master/templates/problem/editorial.html

File: dmoj/settings.py [hunk 3]
```diff
@@ -161,6 +161,9 @@
 FONTAWESOME_CSS = '//maxcdn.bootstrapcdn.com/font-awesome/4.3.0/css/font-awesome.min.css'
 DMOJ_CANONICAL = ''
 
+DMOJ_ENABLE_COMMENTS = True
```
Comment (Xyene) [inline dmoj/settings.py:164]: Small nit, let's also move these up to below the rest of the `DMOJ_` constants/

**Review by Xyene — COMMENTED** [review]
Otherwise LGTM -- could you please rebase and squash your commits? Then we can go ahead and merge this.

File: dmoj/settings.py (hunk 3 above)
Comment (FherStk) [reply dmoj/settings.py]: Done :)

**Comment by FherStk** [comment]
> Otherwise LGTM -- could you please rebase and squash your commits? Then we can go ahead and merge this.

I'm not used to squash and rebase, but I hope I squashed it correctly :)

**Comment by FherStk** [comment]
> Not quite :)
> 
> You want to run `git rebase -i HEAD~10`, which will bring up interactive rebase for the last 10 commits, and change the last few lines to
> 
> ```
> ...
> pick 47863906 new settings options for disabling comments and social
> s eec0b6a9 new settings options for disabling comments and social
> s a528dd54 disable-{social | comments} --> enable-{social | comments}
> s 9f2f5fa5 Fixes for 2304#d
> ```
> 
> `s` short for 'squash', see the big comment in interactive rebase for details. Then you'll be prompted to fix up the commit text for the commit things are being squashed into (`47863906`), after which you'll be able to `--force` push the result onto your branch.

Hi,

Sorry for the delay, I've been a little overwhelmed with my work, but I hope I squashed correctly now. I'm very grateful for all your help and support. 

Thanks again!
Fher

**Review by Xyene — APPROVED** [review]
Thanks for the contribution!

File: resources/libs
Comment (int-y1) [inline resources/libs:1]: why was this submodule updated?

Comment (FherStk) [reply resources/libs:1]: I guess this is related with my tries of smashing commits, because the links displays no changes:
imagen
imagen
